Add css variables rules proposal (#470)
CartoDB/airship: DEVELOPERS.md

We dont want prereleases to be shown in the changelog so **git tags corresponding to a prerelase should be removed once the QA is done**.

> NOTE: You only need to remove it from github and the `changelog` script will prune all removed tags.

## CSS Variables

> To achieve a better customization we use [css variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables). Follow this rules in order to write them in a consistent way.

Reviewer: ```suggestion
To achieve a better customization we use [css variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables). Follow these rules in order to write them in a consistent way.
```

We define two kinds of variables, global and specific:

> - **Global Values:** This variables are declared in `styles/core.scss` under the `:host` selector and define specific values that are availiable on every css scope.

Reviewer: ```suggestion
- **Global Values:** This variables are declared in `styles/core.scss` under the `:root` selector and define specific values that are available on every css scope.
```

> - **Global Variables:** This variables are declared in `styles/core.scss` under the `:host` selector and are availiable on every css scope.

Reviewer: ```suggestion
- **Global Variables:** This variables are declared in `styles/core.scss` under the `:root` selector and are available on every css scope.
```

- **Specific Values:** This variables are declared in the element/component styles and define a value that can be used in different places, usually specific variables. (Same value could be used in the table border and paragraph color)

> - **Specific Variables:** This variables are defined in the stylss of element/component and they only are availiable under the element/component scope.

Reviewer: ```suggestion
- **Specific Variables:** This variables are defined in the styles of element/component and they're only  available under the element/component scope.
```

> The variables should have a less specific one as fallback, so for example when developing the `histogram` widget:

Reviewer: ```suggestion
The variables should have a less specific one as fallback, so for example when developing the `stacked-bar` widget:
```

```css

  /** <style/core/core.scss> The values are initialized **/
  --as--color--primary: white;

  /** <stacked-bar-widget.scss> The values used with chained fallbacks, last fallback is a css-value **/
  --as-stacked-bar-widget--background-color: var(--as--widget__header--background-color, --as--color--primary);
```

> Doing this users can controll specific widgets `--stacked-bar-widget--background-color` share a style for all widgets `--as--widget__header--background-color` or just redefine the default value `--as--color--primary`.

Reviewer: This paragraph is confusing, I rewrote it but I don't know if it's exactly what you meant.

```suggestion
With `--stacked-bar-widget--background-color` users can control specific widgets, with the shared variable `--as--widget__header--background-color` they can control all widgets, or they can just redefine the default value `--as--color--primary`.
```

By the way, the class is `--widget-background-color` right now, not `--as--widget__header--background-color`. I don't know if you plan to change it, just noticed it.

Author: my idea is to change them to follow the rules :)

Reviewer: ![great](https://media.giphy.com/media/jNdw5Qmy5MOpq/giphy.gif)
